source/dub/commandline.d

```diff
@@ -257,6 +257,7 @@ struct CommonOptions {
 	/// Parses all common options and stores the result in the struct instance.
 	void prepare(CommandArgs args)
 	{
+		registry_urls = environment.get("DUB_REGISTRY", null).split(";");
```

Comment:
> It's weird to pass package suppliers to new Dub only to have it add more package suppliers internally.

But it will add more anyway, the default one and the ones from the config file. Also, `Dub` already reads `DUBPATH` for example. So just based on the current state, adding it there would cause a bit less additional confusion - but in general I'd say all of this should be controllable from outside of the `Dub` class and a more principled separation would of course be even better.

Adding a generic way to handle env vars as an alternative to certain command line arguments also open up the possibility for better auto-generated command line help, so that sounds like a good idea. The only thing that should be there, and this is kind of orthogonal to this PR, is a way to pass a custom environment, so that it's possible to opt-out when the package is used as a library. This actually applies to the whole library, though, and not just to the command line module.